amethyst_renderer/src/vertex.rs

```diff
@@ -284,3 +296,36 @@ impl With<TexCoord> for PosNormTangTex {
         format: TexCoord::FORMAT,
     };
 }
+
+
+/// Allows to query specific `Attribute`s of `VertexFormat`
+pub trait Query<T>: VertexFormat {
+    /// Attributes from tuple `T`
+    const QUERIED_ATTRIBUTES: Attributes<'static>;
+}
+
+macro_rules! impl_query {
+    ($($a:ident),*) => {
+        impl<V $(,$a)*> Query<($($a,)*)> for V
+            where V: VertexFormat,
+            $(
+                $a: Attribute,
+                V: With<$a>,
+            )*
+        {
+            const QUERIED_ATTRIBUTES: Attributes<'static> = &[
+                $(
+                    ($a::NAME, <V as With<$a>>::FORMAT),
+                )*
+            ];
+        }
+
+        impl_query!(@ $($a),*);
+    };
+    (@) => {};
+    (@ $head:ident $(,$tail:ident)*) => {
+        impl_query!($($tail),*);
+    };
+}
+
+impl_query!(A, B, C, D);
```

Comment:
Sorry, I really don't know what `Query` is used for and what kind of maximum this macro invocation is responsible for. I lost track of what's going on in our renderer lately :(

Reply:
It just compose some static const slice of named attribute formats from`With` trait.
We have only 4 different attributes right now.
Maybe I can add some more here. There is no cost.

Reply:
Look how passes use it

Reply:
> I lost track of what's going on in our renderer lately :(

You can't keep up with everything 🏃